presto-product-tests/src/main/java/com/facebook/presto/tests/cli/PrestoCliTests.java

```diff
@@ -127,18 +127,71 @@ public void shouldUseCatalogAndSchemaOptions()
     {
         launchPrestoCliWithServerArgument("--catalog", "hive", "--schema", "default", "--execute", "select * from nation;");
         assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
+        presto.waitForWithTimeoutAndKill();
     }
 
     @Test(groups = CLI, timeOut = TIMEOUT)
     public void shouldRunQueryFromFile()
             throws IOException, InterruptedException
     {
-        File temporayFile = File.createTempFile("test-sql", null);
-        temporayFile.deleteOnExit();
-        Files.write("select * from hive.default.nation;\n", temporayFile, UTF_8);
+        try (TempFile file = new TempFile()) {
+            Files.write("select * from hive.default.nation;\n", file.file(), UTF_8);
+
+            launchPrestoCliWithServerArgument("--file", file.file().getAbsolutePath());
+            assertThat(trimLines(presto.readRemainingOutputLines())).containsAll(nationTableBatchLines);
+
+            presto.waitForWithTimeoutAndKill();
```

Comment:
we don't assert anywhere that on success cli exit with `0`.

Reply:
That method does (which is why we assert the other one throws on non-zero)